src/secret.rs

```diff
@@ -0,0 +1,307 @@
+    pub filters: HashMap<T, Vec<T>>,
+}
+
+/// Parameters used in the [Update Secret API](Docker::update_secret())
+///
+/// ## Examples
+///
+/// ```rust
+/// use bollard::secret::UpdateSecretOptions;
+///
+/// UpdateSecretOptions{
+///     version: 1234,
+///     ..Default::default()
+/// };
+/// ```
+#[derive(Debug, Copy, Clone, Default, PartialEq, Serialize)]
+pub struct UpdateSecretOptions {
+    /// The version number of the secret object being updated. This is required to avoid conflicting writes. This version number should be the value as currently set on the secret before the update.
+    pub version: u64,
+}
+
+impl Docker {
+    /// ---
+    ///
+    /// # List Secrets
+    ///
+    /// Returns a list of secrets.
+    ///
+    /// # Arguments
+    ///
+    ///  - Optional [ListSecretsOptions](ListSecretsOptions) struct.
+    ///
+    /// # Returns
+    ///
+    ///  - Vector of [Secret](Secret), wrapped in a Future.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bollard::Docker;
+    /// # let docker = Docker::connect_with_http_defaults().unwrap();
+    /// use bollard::secret::ListSecretsOptions;
+    ///
+    /// use std::collections::HashMap;
+    /// use std::default::Default;
+    ///
+    /// let mut filters = HashMap::new();
+    /// filters.insert("label", vec!["secret-label=label-value"]);
+    ///
+    /// let options = Some(ListSecretsOptions{
+    ///     filters,
+    ///     ..Default::default()
+    /// });
+    ///
+    /// docker.list_secrets(options);
+    /// ```
+    pub async fn list_secrets<T>(
+        &self,
+        options: Option<ListSecretsOptions<T>>,
+    ) -> Result<Vec<Secret>, Error>
+    where
+        T: Into<String> + Eq + Hash + Serialize,
+    {
+        let url = "/secrets";
+
+        let req = self.build_request(
+            url,
+            Builder::new().method(Method::GET),
+            options,
+            Ok(Body::empty()),
+        );
+
+        self.process_into_value(req).await
+    }
+
+    /// ---
+    ///
+    /// # Create Secret
+    ///
+    /// Create new secret on the docker swarm.
+    ///
+    /// # Arguments
+    ///
+    ///  - [SecretSpec](SecretSpec) struct.
+    ///
+    /// # Returns
+    ///
+    ///  - A [String](String) secret Id, wrapped in a Future.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bollard::Docker;
+    /// # use std::default::Default;
+    /// # let docker = Docker::connect_with_http_defaults().unwrap();
+    /// use bollard::secret::SecretSpec;
+    ///
+    /// use base64;
+    ///
+    /// let secret_spec = SecretSpec {
+    ///     name: Some(String::from("secret-name")),
+    ///     data: Some(base64::encode_config("secret-data", base64::URL_SAFE)),
+    ///     ..Default::default()
+    /// };
+    ///
+    /// docker.create_secret(secret_spec);
+    /// ```
+    pub async fn create_secret(&self, secret_spec: SecretSpec) -> Result<String, Error> {
```

Comment:
We should probably keep the response type as `IdResponse` here for forward compatibility reasons, and patch the codegen appropriately. We can report the problem upstream to the moby project and revert the codegen changes when that gets merged.